nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -279,6 +283,72 @@ Costmap2DROS::on_shutdown(const rclcpp_lifecycle::State &)
   return nav2_util::CallbackReturn::SUCCESS;
 }
 
+void
+Costmap2DROS::paramEventCallback(const rcl_interfaces::msg::ParameterEvent::SharedPtr & /*event*/)
```

Comment:
This seems to have alot of code duplication to setup, can this be factored out to handle that?

Also, why are you killing the updater thread? 

Any reason why these cant be individual param updates like the layers?

Reply:
Yes, I agree there is room to factor out some duplicated code.

This callback is essentially a port of the ROS1 dynamic reconfigure callback for the `Costmap2DROS` where the update thread was killed and re-started. As this PR is mostly a direct port, I am open to reasoning out a better design for these callbacks.

Reply:
Ok, I didn’t realize it was doing that. I’ll just go with it then.